overlays/nodejs/cloudwatch/eventRuleMixins.ts

```diff
@@ -0,0 +1,121 @@
+    // event spans a time interval, the service might choose to report the start time, so this value
+    // can be noticeably before the time the event is actually received.
+    time: string;
+
+    // A unique value is generated for every event. This can be helpful in tracing events as they
+    // move through rules to targets, and are processed.
+    id: string;
+
+    // This JSON array contains ARNs that identify resources that are involved in the event.
+    // Inclusion of these ARNs is at the discretion of the service. For example, Amazon EC2 instance
+    // state-changes include Amazon EC2 instance ARNs, Auto Scaling events include ARNs for both
+    // instances and Auto Scaling groups, but API calls with AWS CloudTrail do not include resource
+    // ARNs.
+    resources: string[];
+
+    // A JSON object, whose content is at the discretion of the service originating the event.
+    detail: Record<string, any>;
+}
+
+export type EventRuleEventHandler = lambda.EventHandler<EventRuleEvent, void>;
+
+export class EventRuleEventSubscription extends lambda.EventSubscription {
+    public readonly eventRule: eventRule.EventRule;
+    public readonly target: eventTarget.EventTarget;
+
+    public constructor(
+        name: string, eventRuleOrSchedule: eventRule.EventRule | string, handler: EventRuleEventHandler,
+        args: EventRuleEventSubscriptionArgs, opts?: pulumi.ResourceOptions) {
+
+        super("aws:cloudwatch:EventRuleEventSubscription", name, { }, opts);
+
+        const parentOpts = { parent: this };
+        if (typeof eventRuleOrSchedule === "string") {
+            this.eventRule = new eventRule.EventRule(name, {
+                scheduleExpression: eventRuleOrSchedule
+            },
+            parentOpts);
+        }
+        else {
+            this.eventRule = eventRuleOrSchedule;
+        }
+
+        this.func = lambda.createFunctionFromEventHandler(name, handler, parentOpts);
+
+        this.target = new eventTarget.EventTarget(name, {
+            rule: this.eventRule.name,
+            arn: this.func.arn,
+            targetId: name,
+        }, parentOpts);
+
+        this.permission = new lambda.Permission(name, {
+            action: "lambda:invokeFunction",
+            function: this.func,
+            principal: "events.amazonaws.com",
+            sourceArn: this.eventRule.arn,
+        }, parentOpts);
+    }
+}
+
+// Mixin helpers to create lambda triggers directly from an event rule event.
+
+declare module "./eventRule" {
+    interface EventRule {
+        /**
+         * Creates a new subscription to events fired from this EventRule to the handler provided, along
+         * with options to control the behavior of the subscription.
+         */
+        onEvent(name: string, handler: EventRuleEventHandler,
+                args?: EventRuleEventSubscriptionArgs, opts?: pulumi.ResourceOptions): EventRuleEventSubscription;
```

Comment:
bikeshedding. how does .onEvent sound to people here?   we could say .onEventFired, but i don't think there's high value in changing hte name.